cli/src/commands/util/exec.rs

```diff
@@ -76,23 +78,32 @@ pub fn cmd_util_exec(
     _command: &CommandHelper,
     args: &UtilExecArgs,
 ) -> Result<(), CommandError> {
-    let status = std::process::Command::new(&args.command)
-        .args(&args.args)
-        .status()
-        .map_err(|err| {
-            user_error_with_message(
-                format!("Failed to execute external command '{}'", &args.command),
-                err,
-            )
-        })?;
-    if !status.success() {
-        let error_msg = if let Some(exit_code) = status.code() {
-            format!("External command exited with {exit_code}")
-        } else {
-            // signal
-            format!("External command was terminated by: {status}")
-        };
-        return Err(user_error(error_msg));
+    let mut cmd = std::process::Command::new(&args.command);
+    cmd.args(&args.args);
+
+    // If it's supported on the current platform, perform an execv.
+    #[cfg(unix)]
+    let status: std::io::Result<ExitStatus> =
+        Err(std::os::unix::process::CommandExt::exec(&mut cmd));
+    #[cfg(not(unix))]
+    let status = cmd.status();
+
```

Comment:
Is it important to replace the current process?

If the problem is just exit code (of non-signal termination), it's simpler to spawn, wait, and exit with the same status code.

Reply:
I'm concerned more about different behaviors between platforms. I think we should rely on Rust's platform abstraction so long as it works. Signal termination won't be detected by the interactive shell, but I think that's okay if we print that and exit with non-zero status.